spec/requests/shared_examples/api/v4/shared_spec_for_sync_requests.rb

```diff
@@ -158,4 +158,32 @@ def assert_sync_success(response, request_process_token)
       expect(JSON(response.body)[response_key].count).to eq(1)
     end
   end
+
+  context "translations are correctly inferred" do
+    it "sets locale correctly" do
+      expect(I18n).to receive(:with_locale).with("bn-IN")
+
+      get sync_route, params: {}, headers: headers.merge({"Accept-Language" => "bn-IN"})
+    end
+
+    it "defaults to first primary language when it cannot find region" do
+      expect(I18n).to receive(:with_locale).with("en")
+
+      get sync_route, params: {}, headers: headers.merge({"Accept-Language" => "en-IN"})
```

Comment:
`en-IN` should match to itself, and not `en`

Reply:
updated test.